lib/builder.js

```diff
@@ -246,16 +246,26 @@ function executeConfigFile(saveForReset, ignoreBaseURL, source) {
   if (saveForReset)
     this.resetConfig = source;
 
-  var builder = this;
-  var curSystem = global.System;
-  var configSystem = global.System = {
-    config: function(cfg) {
-      builder.config(cfg, ignoreBaseURL);
+  // Save existing System.config function.
+  var systemConfigFunc = global.System.config;
```

Comment:
We can't assume that it is the global system.

Try using rather - `var System = this.loader`, so that it will naturally be in the function scope below anyway, unless a user explicitly writes `window.System.config(...)`. We can swap out the global otherwise of course.